cmake/dts.cmake

```diff
@@ -120,7 +120,7 @@ if(SUPPORTS_DTS)
   # intermediary file *.dts.pre.tmp. Also, generate a dependency file
   # so that changes to DT sources are detected.
   execute_process(
-    COMMAND ${CMAKE_C_COMPILER}
+    COMMAND gcc
```

Comment:
This is not OK.

If this compiler can't preprocess DT sources then you need to introduce an abstraction like

"CMAKE_C_COMPILER_fully_featured"

or similar that can preprocess DT sources.

Then normally we can set this variable to CMAKE_C_COMPILER, and as an exception do something else for
this compiler. But the best thing would be to have this compiler be able to preprocess DTS.

Reply:
It may be a bug of metaware compiler,  the options are ok, but emitt error msg, i am checking with snps compiler team.